test/component_test.rb

```diff
@@ -114,6 +116,40 @@ class ComponentTest < ActiveSupport::TestCase
     assert_equal [], component.foos
   end
 
+  test "initialize with given attribute and successfull validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+    end
+
+    assert_nothing_raised { component_class.new(:view, foo: "bar") }
+  end
+
+  test "initialize without attribute and failing validation" do
+    component_class = Class.new(Components::Component) do
+      attribute :foo
+      validates :foo, presence: true
+
+      # supplies a name argument for active validations on anonymous classes
+      def self.model_name
+        ActiveModel::Name.new(self, nil, "temp")
+      end
```

Comment:
Will this be a problem when putting validations on element attributes as well? Since they are anonymous classes. Could we have a test case for this as well? Something like:

```ruby
component_class = Class.new(Components::Component) do
  element :foo do
    attribute :bar
    validates :bar, presence: true
  end
end
```

Reply:
Good point.

Reply:
I defined an ActiveModel::Name for the element and component class, which enables validations on anonymous class initialization.